homeassistant/components/spider/__init__.py

```diff
@@ -1,57 +1,93 @@
 """Support for Spider Smart devices."""
-from datetime import timedelta
+import asyncio
 import logging
 
 from spiderpy.spiderapi import SpiderApi, UnauthorizedException
 import voluptuous as vol
 
+from homeassistant.config_entries import SOURCE_IMPORT
 from homeassistant.const import CONF_PASSWORD, CONF_SCAN_INTERVAL, CONF_USERNAME
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.discovery import load_platform
 
-_LOGGER = logging.getLogger(__name__)
-
-DOMAIN = "spider"
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN, PLATFORMS
 
-SPIDER_COMPONENTS = ["climate", "switch"]
-
-SCAN_INTERVAL = timedelta(seconds=120)
+_LOGGER = logging.getLogger(__name__)
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
             {
                 vol.Required(CONF_PASSWORD): cv.string,
                 vol.Required(CONF_USERNAME): cv.string,
-                vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL): cv.time_period,
+                vol.Optional(
+                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
+                ): cv.time_period,
             }
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-def setup(hass, config):
-    """Set up Spider Component."""
+def _spider_startup_wrapper(entry):
+    """Startup wrapper for spider."""
+    try:
+        api = SpiderApi(
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            entry.data[CONF_SCAN_INTERVAL],
+        )
+        return api
+    except UnauthorizedException:
```

Comment:
We don't want to return None implicitly here. Probably let the exception leak out to `async_setup_entry` and `return False` there in that case.